nixos/modules/system/boot/luksroot.nix

```diff
@@ -268,6 +287,11 @@ let
 
             if [ $? == 0 ]; then
                 opened=true
+                ${if luks.reusePassphrases then ''
+                  # We don't rm here because we might reuse it for the next device
+                '' else ''
+                  rm -f /crypt-ramfs/k_user
```

Comment:
how does this file exist if it's created on L269 which is behind an opposite conditional?

Reply:
Nice catch! I agree that at this point the conditional seems unnecessary. This logic was initially introduced in [a9d69a74d6e](https://github.com/NixOS/nixpkgs/commit/a9d69a74d6edb6bcca29b1189d4bc3b203ecaf25) for the function `do_open_passphrase` along with the script `cryptsetup-askpass`, which performs the passphrase redirection without a conditional. The script was referenced in the comment "ask cryptsetup-askpass" in the function but not actually used there. Later, the conditional in question was duplicated as-is in [147621f7db6](https://github.com/NixOS/nixpkgs/commit/147621f7db67344b781c6626b88ff6d0bad7aa70) for function `do_open_gpg_card`. Noticing this, I simply followed suit and duplicated it here for function `do_open_yubikey`. As a newcomer considering this file’s history, this somehow seemed the safer bet.